sled-agent/src/nexus.rs

```diff
@@ -39,27 +39,57 @@ pub struct LazyNexusClient {
 }
 
 impl LazyNexusClient {
-    pub fn new(log: Logger, addr: Ipv6Addr) -> Result<Self, ResolveError> {
+    pub fn new_from_subnet(
```

Comment:
Can we document what this does?  (I find it very non-obvious what it would mean to construct a Nexus client from an "ip" that we refer to as a subnet.)  I gather what we're really saying is: given this IP address, which could be _any_ IP in the AZ, use a hardcoded policy to find the DNS servers ("the AZ subnet is a /56 and the first N addresses are DNS servers") and use those to find Nexus.

Reply:
I'm deprecating `LazyNexusClient`, per your suggestion in a different comment, but I can clean up the documentation in `internal-dns/src/resolver.rs`, where this is also referenced.

Reply:
Done in fb74e5c9e